kart/annotations/db.py

```diff
@@ -0,0 +1,51 @@
+import contextlib
+import json
+import threading
+from sqlalchemy import Column, Integer, Text
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+from kart.sqlalchemy.create_engine import sqlite_engine
+
+Base = declarative_base()
+
+
+class KartAnnotation(Base):
+    __tablename__ = "kart_annotations"
+    id = Column(Integer, nullable=False, primary_key=True)
+    # TODO: look into what indexes we should have here
+    object_type = Column(Text, nullable=False)
```

Comment:
does `object_type` provide any useful benefit?

Reply:
Maybe/maybe not. I expect to annotate trees at some stage (to provide things like full-dataset feature counts).

If we were to have annotations for trees _and_ commits, they'll be opaque OIDs. They won't conflict (because OIDs don't) but it's still nice to be able to see what they are.

Reply:
For how often it'd be used, is `git cat-file -t {oid}` (or `... | git cat-file --batch-check=%(objecttype)` in bulk) enough?